### PR TITLE
[receiver/mongodbatlas] Adds additional disk & process metrics

### LINKED ISSUE #36525: [receiver/mongodbatlas] Add additional collected metrics
### Component(s)

receiver/mongodbatlas

### Is your feature request related to a problem? Please describe.

There are a number of metrics available that are useful (and that may already be collected from the API as far as I can tell), but not currently being made into metrics. Adding these metrics would help to have parity with the monitoring graphs that MongoDB Atlas has to offer natively. 

The list is:
* `mongodbatlas.process.cache.ratio`
* `mongodbatlas.disk.partition.throughput`
* `mongodbatlas.disk.partition.queue.depth`

### Describe the solution you'd like

1. Add new metrics:

`mongodbatlas.process.cache.ratio` to be placed under a Process resource.

`mongodbatlas.disk.partition.throughput`
`mongodbatlas.disk.partition.queue.depth` to be placed under a Process/Disk resource.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by StefanKurek**
#### Description
Adds additional disk and process metrics to flesh out available monitoring metrics from the API. No new API calls were needed as we were already getting this data already.

#### Link to tracking issue
Fixes #36525

#### Testing
New tests generated and ran.

#### Documentation
New documentation generated.

### CODE REVIEW

File: .chloggen/mongodbatlasreceiver-metric-adds.yaml
Comment (schmikei) [inline .chloggen/mongodbatlasreceiver-metric-adds.yaml:1]: I don't think the content of this file is what you intended?

File: receiver/mongodbatlasreceiver/README.md
Comment (schmikei) [inline receiver/mongodbatlasreceiver/README.md:11]: Not super familiar but just following this link right now does not take me to appropriate beta description located here: https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/component-stability.md#beta

File: receiver/mongodbatlasreceiver/internal/metric_conversion.go
Comment (schmikei) [inline receiver/mongodbatlasreceiver/internal/metric_conversion.go:51]:
```suggestion
		case "DISK_PARTITION_THROUGHPUT_WRITE", "DISK_PARTITION_THROUGHPUT_READ":
```

Would it not be simpler to not relying on the fallthrough? I at least think it reads a tad bit better

File: receiver/mongodbatlasreceiver/internal/metric_conversion.go
Comment (schmikei) [inline receiver/mongodbatlasreceiver/internal/metric_conversion.go:60]: Will probably need to test to make sure matching timestamps is sufficent enough to combine datapoints. I may have to poke around the API to make sure that is a valid assumption at this step.

**Review by schmikei — CHANGES_REQUESTED** [review]

File: receiver/mongodbatlasreceiver/README.md
Comment (StefanKurek) [reply receiver/mongodbatlasreceiver/README.md:11]: @schmikei hmm, I don't think I changed this manually (just had it generated...). But I can double check.

File: .chloggen/mongodbatlasreceiver-metric-adds.yaml
Comment (StefanKurek) [reply .chloggen/mongodbatlasreceiver-metric-adds.yaml:1]: @schmikei nope, forgot to push up those changes!. should be good now though

File: receiver/mongodbatlasreceiver/internal/metric_conversion.go
Comment (StefanKurek) [reply receiver/mongodbatlasreceiver/internal/metric_conversion.go:51]: @schmikei can you tell I haven't been writing go consistently in awhile...Looks like a good change to me!

File: receiver/mongodbatlasreceiver/internal/metric_conversion.go
Comment (StefanKurek) [reply receiver/mongodbatlasreceiver/internal/metric_conversion.go:60]: @schmikei I have done some manual spot checking and it has looked good so far (multiple datapoints coming through in a single collection, and the totals being created properly).

Outside of that I'll leave it up to you. I can rip this out if need be, as ultimately it was a "nice to have" although hopefully it doesn't come to it.

File: receiver/mongodbatlasreceiver/README.md
Comment (StefanKurek) [reply receiver/mongodbatlasreceiver/README.md:11]: yeah...mdatagen makes the change you see here...

**Review by schmikei — APPROVED** [review]

**Comment by schmikei** [comment]
@djaglowski as a codeowner I've tested this change and I think it should be good to merge at this point

**Review by djaglowski — APPROVED** [review]

**Pull request merged**
